bokeh/core/validation/check.py

```diff
@@ -85,24 +88,24 @@ def silenced(warning: int) -> None:
         silence(warning, False)
 
 def check_integrity(models):
-    ''' Apply validation and integrity checks to a collection of Bokeh models.
+    ''' Collect all warnings associated with a collection of Bokeh models.
 
     Args:
         models (seq[Model]) : a collection of Models to test
 
     Returns:
-        None
+        dict(error=[], warning=[]): A dictionary of all warning and error messages
 
-    This function will emit log warning and error messages for all error or
-    warning conditions that are detected. For example, layouts without any
-    children will trigger a warning:
+    This function will return a dictionary containing all errors or
+    warning conditions that are detected. For example, layouts without
+    any children will add a warning to the dictionary:
 
     .. code-block:: python
 
         >>> empty_row = Row
 
         >>> check_integrity([empty_row])
-        W-1002 (EMPTY_LAYOUT): Layout has no children: Row(id='2404a029-c69b-4e30-9b7d-4b7b6cdaad5b', ...)
+        dict(warning = [(1002, EMPTY_LAYOUT, Layout has no children, Row(id='2404a029-c69b-4e30-9b7d-4b7b6cdaad5b', ...))])
```

Comment:
I think this would, read better in pretty-printed "literal" form
```suggestion
        {
            "warning": [
                (1002, EMPTY_LAYOUT, Layout has no children, Row(id='2404a029-c69b-4e30-9b7d-4b7b6cdaad5b', ...),
                ...
            ],
            "error": [...]
        }             
```